image_classification/HVT/main_multi_gpu.py

```diff
@@ -0,0 +1,611 @@
+            logger_name='local_logger')
+    # overall logger
+    if local_rank == 0:
+        master_logger = get_logger(
+            filename=os.path.join(config.SAVE, 'log.txt'),
+            logger_name='master_logger')
+        master_logger.info(f'\n{config}')
+    else:
+        master_logger = None
+    local_logger.info(f'----- world_size = {world_size}, local_rank = {local_rank}')
+    if local_rank == 0:
+        master_logger.info(f'----- world_size = {world_size}, local_rank = {local_rank}')
+    
+    # STEP 1: Create model
+    model = build_model(config)
+    # define model ema
+    model_ema = None
+    if not config.EVAL and config.TRAIN.MODEL_EMA and local_rank == 0:
+        model_ema = ModelEma(model, decay=config.TRAIN.MODEL_EMA_DECAY)
+    model = paddle.DataParallel(model)
+
+    # STEP 2: Create train and val dataloader
+    dataset_train, dataset_val = args[1], args[2]
+    # Create training dataloader
+    if not config.EVAL:
+        dataloader_train = get_dataloader(config, dataset_train, 'train', True)
+        total_batch_train = len(dataloader_train)
+        local_logger.info(f'----- Total # of train batch (single gpu): {total_batch_train}')
+        if local_rank == 0:
+            master_logger.info(f'----- Total # of train batch (single gpu): {total_batch_train}')
+    # Create validation dataloader
+    dataloader_val = get_dataloader(config, dataset_val, 'test', True)
+    total_batch_val = len(dataloader_val)
+    local_logger.info(f'----- Total # of val batch (single gpu): {total_batch_val}')
+    if local_rank == 0:
+        master_logger.info(f'----- Total # of val batch (single gpu): {total_batch_val}')
+
+    # STEP 3: Define Mixup function
+    mixup_fn = None
+    if config.TRAIN.MIXUP_PROB > 0 or config.TRAIN.CUTMIX_ALPHA > 0 or config.TRAIN.CUTMIX_MINMAX is not None:
+        mixup_fn = Mixup(mixup_alpha=config.TRAIN.MIXUP_ALPHA,
+                         cutmix_alpha=config.TRAIN.CUTMIX_ALPHA,
+                         cutmix_minmax=config.TRAIN.CUTMIX_MINMAX,
+                         prob=config.TRAIN.MIXUP_PROB,
+                         switch_prob=config.TRAIN.MIXUP_SWITCH_PROB,
+                         mode=config.TRAIN.MIXUP_MODE,
+                         label_smoothing=config.TRAIN.SMOOTHING,
+                         num_classes=config.MODEL.NUM_CLASSES)
+
+    # STEP 4: Define criterion
+    if config.TRAIN.MIXUP_PROB > 0.:
+        criterion = SoftTargetCrossEntropyLoss()
+    elif config.TRAIN.SMOOTHING:
+        criterion = LabelSmoothingCrossEntropyLoss()
+    else:
+        criterion = nn.CrossEntropyLoss()
+    # only use cross entropy for val
+    criterion_val = nn.CrossEntropyLoss()
+
+
+    # 5. Create Teacher model
+    teacher_model = None
+
+    # STEP 5: Define optimizer and lr_scheduler
+    # set lr according to batch size and world size (hacked from Swin official code and modified for CSwin)
+    if config.TRAIN.LINEAR_SCALED_LR is not None:
```

Comment:
There is no LINEAR_SCALED_LR in  config.py, please add this keyword in config.py